### PR TITLE
17 pocketguide

### LINKED ISSUE #17: pocketguide fragment
find a way to not repeat the cocktail pocket guide fragment in the docs and style it properly.
decision with Karl on 25th Aug morning: create a dedicated pocketguide document that the author can adjust.
author has to add a pocketguide custom block to every page where he wants to include the pocket guide.
the pocketguide will be added on the cocktail pages within their template, so the importer of the cocktail pages shall simply remove the picketguide from it.

### PR DESCRIPTION
**Pull request by RemoLiechti**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #17

Test URLs:
- Before: https://main--sazerac-wheatleyvodka--hlxsites.hlx.page/
- After: https://17-pocketguide--sazerac-wheatleyvodka--hlxsites.hlx.page/drafts/pocketguidepreview

### CODE REVIEW

File: blocks/pocketguide/pocketguide.css [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    position:relative;
+    padding-top:136px;
+    border-radius:10px;
+    border:1px solid #7f5732;
+    margin:100px 15px;
+    min-width: 200px;
+}
+
+.pocketguide>*>*>h2 {
+    text-transform: none;
```
Comment (cziegeler) [inline blocks/pocketguide/pocketguide.css:12]: this should not be needed anymore, styles.css does not set text transform for h2

File: blocks/pocketguide/pocketguide.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * loads and decorates the pocketguide
+ * @param {Element} block The pocketguide block element
+ */
+export default async function decorate(block) {
+  const cfg = readBlockConfig(block);
+  block.textContent = '';
+
+  // fetch footer content
+  const footerPath = cfg.footer || '/templates/pocketguide';
+  const resp = await fetch(`${footerPath}.plain.html`, window.location.pathname.endsWith('/templates/pocketguide') ? { cache: 'reload' } : {});
```
Comment (cziegeler) [inline blocks/pocketguide/pocketguide.js:13]: I think the footer and footer cfg is not needed here, so you can go for the hard-coded template path for now.

File: blocks/pocketguide/pocketguide.css (hunk 1 above)
Comment (RemoLiechti) [reply blocks/pocketguide/pocketguide.css:12]: true, sorry didnt see this copy&waste part

File: blocks/pocketguide/pocketguide.js (hunk 2 above)
Comment (RemoLiechti) [reply blocks/pocketguide/pocketguide.js]: true, sorry didnt see this copy&waste part

File: blocks/pocketguide/pocketguide.css (hunk 1 above)
Comment (RemoLiechti) [reply blocks/pocketguide/pocketguide.css:12]: it does, see styles.css:285

File: blocks/pocketguide/pocketguide.css [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    text-align:center;
+    position:relative;
+    padding-top:136px;
+    border-radius:10px;
+    border:1px solid #7f5732;
+    margin:100px 15px;
+    min-width: 200px;
+}
+
+.pocketguide>*>*>h2 {
```
Comment (karlpauls) [inline blocks/pocketguide/pocketguide.css:11]: I think here you can just use `.pocketguide h2`.

File: blocks/pocketguide/pocketguide.css (hunk 1 above)
Comment (karlpauls) [reply blocks/pocketguide/pocketguide.css:12]: I think @cziegeler removed that one - hence, we can remove here as well for now.

File: blocks/pocketguide/pocketguide.css [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    text-transform: none;
+}
+
+.pocketguide>div>div>p:nth-child(4)>picture{
```
Comment (karlpauls) [inline blocks/pocketguide/pocketguide.css:15]: Are we expecting more than one picture? If not, I think it makes more sense to simplify this selector to just target the picture.

File: blocks/pocketguide/pocketguide.css [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    height:190.8px;
+    width:127.4px;
+    right:calc(50% - 63.7px);
+    top:-60px;
+    position:absolute;
+    transform:rotate(13deg)
+}
+
+@media only screen and (min-width:1000px) {
```
Comment (karlpauls) [inline blocks/pocketguide/pocketguide.css:24]: I think you are missing another breakpoint at 1200px.

File: blocks/pocketguide/pocketguide.css [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    .pocketguide>*>*>h2 {
```
Comment (karlpauls) [inline blocks/pocketguide/pocketguide.css:25]: Simplify selector - see above

File: blocks/pocketguide/pocketguide.css [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    .pocketguide {
+        height:405px;
+        padding:0 0 0 40px;
+        text-align:left;
+        display:flex;
+        flex-direction:column;
+        justify-content:center;
+    }
+
+    .pocketguide>div>div>p:nth-child(4)>picture{
```
Comment (karlpauls) [inline blocks/pocketguide/pocketguide.css:38]: Simplify select - see above

File: blocks/pocketguide/pocketguide.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+ */
+export default async function decorate(block) {
+  block.textContent = '';
+
+  const resp = await fetch('/templates/pocketguide.plain.html', window.location.pathname.endsWith('/templates/pocketguide') ? { cache: 'reload' } : {});
+
+  if (resp.ok) {
+    const html = await resp.text();
+
+    // decorate footer DOM
```
Comment (karlpauls) [inline blocks/pocketguide/pocketguide.js:13]: This comment looks wrong - maybe copy'n'paste mistake?

**Review by karlpauls — CHANGES_REQUESTED** [review]
Can you please publish (or at least preview) the template? It doesn't seem to work for me like this - I would like to see the pocket guide. Additionally, it would be good to change the check url of the PR to actually target a page that contains the pocket guide.

**Comment by RemoLiechti** [comment]
> Can you please publish (or at least preview) the template? It doesn't seem to work for me like this - I would like to see the pocket guide. Additionally, it would be good to change the check url of the PR to actually target a page that contains the pocket guide.

Seems like the pocketguide template was unpublished, yesterday it was visible. I republished it. I will change  the check url in this PR then.

File: blocks/pocketguide/pocketguide.css (hunk 5 above)
Comment (RemoLiechti) [reply blocks/pocketguide/pocketguide.css:24]: dont think so, there's breakpoints for 1000 related to pocketguide only. 
however, there's BPs for 1138, 1200 and 1800 for the home, hero and desktop banner. not related to pocketguide though.

Comment (karlpauls) [reply blocks/pocketguide/pocketguide.css:20]: The text-align left seems to happen after 1200px - not at 1000px already, no?

File: blocks/pocketguide/pocketguide.css [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+.pocketguide picture{
+    height:190.8px;
+    width:127.4px;
+    right:calc(50% - 63.7px);
+    top:-60px;
+    position:absolute;
+    transform:rotate(13deg)
+}
+
+
```
Comment (RemoLiechti) [inline blocks/pocketguide/pocketguide.css:20]: the BP in the middle at 1000 is basically triggered by font size changes only, so the pocketguide does not really care at this point in time. when reaching 1152, there's more to do, not just font sizes.

**Comment by RemoLiechti** [comment]
so now we have

BP 1152+
image


1000-1152
image


below 1000
image

**Review by karlpauls — APPROVED** [review]
LGTM - thanks!

**Pull request merged**